File: src/main/java/club/sk1er/mods/core/universal/UniversalMouse.java
package club.sk1er.mods.core.universal;

//#if FABRIC
//$$ import net.minecraft.client.Mouse;
//#elseif MC<=11202
import org.lwjgl.input.Mouse;
//#endif

public class UniversalMouse {
    public static int getTrueX() {
        //#if FABRIC
        //$$ return (int) UniversalMinecraft.getMinecraft().mouse.getX();
        //#elseif MC>=11502
        //$$ return (int) UniversalMinecraft.getMinecraft().mouseHelper.getMouseX();
        //#else
        return Mouse.getX();
        //#endif
    }

    public static int getScaledX() {
        int trueX = getTrueX();
        UniversalResolutionUtil instance = UniversalResolutionUtil.getInstance();
        return trueX * instance.getScaledWidth() / Math.max(1, instance.getWindowWidth());
    }

    public static int getTrueY() {
        //#if FABRIC
        //$$ return (int) UniversalMinecraft.getMinecraft().mouse.getX();
        //#elseif MC>=11502
        //$$ return UniversalResolutionUtil.getInstance().getWindowHeight() - (int) UniversalMinecraft.getMinecraft().mouseHelper.getMouseY();
        //#else
        return Mouse.getY();
        //#endif
    }

    public static int getScaledY() {
        UniversalResolutionUtil instance = UniversalResolutionUtil.getInstance();
        return getTrueY() * instance.getScaledHeight() / Math.max(1, instance.getWindowHeight());
    }
}

File: src/main/kotlin/club/sk1er/mods/core/universal/UniversalMouse.kt
package club.sk1er.mods.core.universal

//#if MC<=11202
import org.lwjgl.input.Mouse
//#endif

import kotlin.math.max

object UniversalMouse {
    @JvmStatic
    fun getTrueX(): Double {
        //#if FABRIC
        //$$ return UniversalMinecraft.getMinecraft().mouse.x
        //#elseif MC>=11502
        //$$ return UniversalMinecraft.getMinecraft().mouseHelper.mouseX
        //#else
        return Mouse.getX().toDouble()
        //#endif
    }

    @JvmStatic
    fun getScaledX(): Double {
        return getTrueX() * UniversalResolutionUtil.scaledWidth / max(1, UniversalResolutionUtil.windowWidth)
    }

    @JvmStatic
    fun getTrueY(): Double {
        //#if FABRIC
        //$$ return UniversalMinecraft.getMinecraft().mouse.y
        //#elseif MC>=11502
        //$$ return UniversalMinecraft.getMinecraft().mouseHelper.mouseY
        //#else
        return Mouse.getY().toDouble()
        //#endif
    }

    @JvmStatic
    fun getScaledY(): Double {
        return getTrueY() * UniversalResolutionUtil.scaledHeight / max(1, UniversalResolutionUtil.windowHeight)
    }
}
